tests/integration/api/test_geti_interaction.py

```diff
@@ -0,0 +1,209 @@
+            work_dir=self.tmp_path,
+            data_root=self.arrow_file_path,
+        )
+
+    def test_model_and_data_module(self):
+        """Test the instance type of the model and the datamodule."""
+        assert isinstance(self.engine.model, OTXModel)
+        assert isinstance(self.engine.datamodule, OTXDataModule)
+
+    def test_training(self):
+        """Test the training process."""
+        max_epochs = 2
+        self.train_kwargs["max_epochs"] = max_epochs
+        train_metric = self.engine.train(**self.train_kwargs)
+        assert len(train_metric) > 0
+        assert self.engine.checkpoint
+
+    def test_predictions(self):
+        """Test the prediction process. This is way to check that the model is valid."""
+        predictions = self.engine.predict()
+        assert predictions is not None
+        assert len(predictions) > 0
+
+    def test_export_and_infer_onnx(self):
+        """Test exporting the model to ONNX."""
+        for precision in [OTXPrecisionType.FP16, OTXPrecisionType.FP32]:
+            exported_path = self.engine.export(
+                export_format=OTXExportFormatType.ONNX,
+                export_precision=precision,
+                explain=(precision == OTXPrecisionType.FP32),
+                export_demo_package=False,
+            )
+            export_dir = exported_path.parent
+            assert export_dir.exists()
+
+            # Test how model is loaded by Model API
+            if self.mapi_model_class is not None:
+                onnx_path = export_dir / "exported_model.onnx"
+                mapi_model = self.mapi_model_class.create_model(onnx_path)
+                assert mapi_model is not None
+
+                predictions = mapi_model(self.image)
+                assert predictions is not None
+                assert len(predictions.top_labels) > 0
+
+            exported_path.unlink(missing_ok=True)
+
+    def test_export_and_infer_openvino(self):
+        """Test exporting the model to OpenVINO."""
+        for precision in [OTXPrecisionType.FP16, OTXPrecisionType.FP32]:
+            exported_path = self.engine.export(
+                export_format=OTXExportFormatType.OPENVINO,
+                export_precision=precision,
+                explain=(precision == OTXPrecisionType.FP32),
+                export_demo_package=True,
+            )
+            export_dir = exported_path.parent
+            assert export_dir.exists()
+
+            # Test Model API model
+            if self.mapi_model_class is not None:
+                ov_export_dir = self.tmp_path / "ov_export"
+                ov_export_dir.mkdir(parents=True, exist_ok=True)
+                unzip_exportable_code(
+                    work_dir=self.tmp_path,
+                    exported_path=exported_path,
+                    dst_dir=ov_export_dir,
+                )
+                xml_path = ov_export_dir / "exported_model.xml"
+                mapi_model = self.mapi_model_class.create_model(xml_path)
```

Comment:
There's no need to specify a particular class: `Model.create(...)` should work

Reply:
how will it differentiate between e.g. cls and det models?

Reply:
That information is already embedded to `.xml` files